src/config.py

```diff
@@ -117,47 +116,6 @@ def interpolate(value: str) -> str:
             raise KeyError(f"Environment variable '{missing_var}' not found. ") from e
 
 
-def parse_query_parameters(params: list[dict[str, Any]]) -> list[QueryParameter]:
```

Comment:
Its cool that this is moved, but Its only used in this file and its part of the config parsing. Moving this makes sense, however now I would argue that it should become a private static method on the DuneSource class used to parse the `param: dict[list[str, Any]]`.

Reply:
Then the same would apply to the following functions in `src/sources/dune.py`:

```
_parse_decimal_type
_reformat_varbinary_columns
_reformat_unknown_columns
_handle_column_types
dune_result_to_df
```

Reply:
It doesn't have to be a private static class method.